runtime/compilers/rillv1beta/project.go

```diff
@@ -117,3 +117,17 @@ func (c *Codec) DeleteSource(ctx context.Context, name string) (string, error) {
 	}
 	return p, nil
 }
+
+func (c *Codec) ProjectConfig(ctx context.Context) (*ProjectConfig, error) {
+	content, err := c.Repo.Get(ctx, c.InstanceID, "rill.yaml")
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
```

Comment:
To make clear we don't get `nil` pointer accesses after this, it would be better to explicitly return an empty `ProjectConfig{Env: make(map[string]string)}` when `os.IsNotExist(err)`.